spec/controllers/controller_spec.rb

```diff
@@ -85,6 +85,39 @@
       end
     end
 
+    describe '#login!' do
+      context 'when succeeds' do
+        before do
+          expect(User).to receive(:authenticate).with('bla@bla.com', 'secret') { |&block| block.call(user, nil) }
```

Comment:
It looks like this test calls User.authenticate directly rather than the login method. The controller spec should test the controller methods.

Reply:
Ah, wait, never mind. It's been a minute since I've looked at Ruby. This is overwriting the authenticate method for User during the duration of the test.

Is this necessary?

Reply:
I duplicated this code from the `#login` tests [above](https://github.com/Sorcery/sorcery/pull/332/files/9ebaf3166c99b63aa2f3c14dc48a4f4c5b537387#diff-1f60346fac30942dff621cab795a97495a4117679cec4ba682995891a0628e3cL55), but i can remove it if it's unnecessary.

Reply:
:facepalm: 

That's fine, then. I think it would be better to be consistent and fix it all at once later.